src/main/scala/pbdirect/PBMessageWriter.scala

```diff
@@ -28,17 +28,38 @@ trait PBMessageWriterImplicits {
     }
   }
 
-  implicit def prodWriter[A, R <: HList, Anns <: HList, ZWI <: HList, ZWFI <: HList](
+  object zipWithModifiers extends Poly2 {
+    implicit def annotatedCase[A] = at[(FieldIndex, A), Some[pbUnpacked]] {
+      case ((fieldIndex, value), Some(annotation)) =>
+        (fieldIndex, value, FieldModifiers(unpacked = true))
+    }
+    implicit def unannotatedCase[A] = at[(FieldIndex, A), None.type] {
+      case ((fieldIndex, value), None) =>
+        (fieldIndex, value, FieldModifiers(unpacked = false))
+    }
+  }
+
+  implicit def prodWriter[
```

Comment:
This method signature is turning into a work of art. If we add support for more annotations such as `@pbFixedWidth` and `@pbSigned`, we'll need to do even more zipping. Let's refactor it at that point.